test/config/config.ts

```diff
@@ -1,98 +1,346 @@
-      max: Number(env.SEQ_IBMI_POOL_MAX || env.SEQ_POOL_MAX || env.SEQ_POOL_MAX || 5),
-      idle: Number(env.SEQ_IBMI_POOL_IDLE || env.SEQ_POOL_IDLE || 3000),
-    },
-    dialectOptions: {
-      odbcConnectionString: env.SEQ_IBMI_CONN_STR,
+    native: { port: 8471 },
+    default: {
+      database: env.SEQ_IBMI_DB || env.SEQ_DB,
+      username: env.SEQ_IBMI_USER || env.SEQ_USER,
+      password: env.SEQ_IBMI_PW || env.SEQ_PW,
+      pool: {
+        max: Number(env.SEQ_IBMI_POOL_MAX || env.SEQ_POOL_MAX || env.SEQ_POOL_MAX || 5),
+        idle: Number(env.SEQ_IBMI_POOL_IDLE || env.SEQ_POOL_IDLE || 3000),
+      },
+      dialectOptions: {
+        odbcConnectionString: env.SEQ_IBMI_CONN_STR,
+      },
     },
   },
 };
+
+//   ██████╗ ██████╗ ███╗   ██╗███████╗██╗ ██████╗     ███████╗██╗   ██╗███╗   ██╗ ██████╗
+//  ██╔════╝██╔═══██╗████╗  ██║██╔════╝██║██╔════╝     ██╔════╝██║   ██║████╗  ██║██╔════╝
+//  ██║     ██║   ██║██╔██╗ ██║█████╗  ██║██║  ███╗    █████╗  ██║   ██║██╔██╗ ██║██║
+//  ██║     ██║   ██║██║╚██╗██║██╔══╝  ██║██║   ██║    ██╔══╝  ██║   ██║██║╚██╗██║██║
+//  ╚██████╗╚██████╔╝██║ ╚████║██║     ██║╚██████╔╝    ██║     ╚██████╔╝██║ ╚████║╚██████╗
+//   ╚═════╝ ╚═════╝ ╚═╝  ╚═══╝╚═╝     ╚═╝ ╚═════╝     ╚═╝      ╚═════╝ ╚═╝  ╚═══╝ ╚═════╝
+//
+
+export function Config(dialect: string | undefined): Options {
+
+  if (dialect === undefined) {
+    return {};
+  }
+
+  let _fullString: string;
+  let dialectName: string;
+  let dialectVersion: string;
+  let didEncounterSpecialCase = false;
+
+  //  ┌─┐┌─┐┌─┐┌─┐┬┌─┐┬    ┌─┐┌─┐┌─┐┌─┐┌─┐
+  //  └─┐├─┘├┤ │  │├─┤│    │  ├─┤└─┐├┤ └─┐
+  //  └─┘┴  └─┘└─┘┴┴ ┴┴─┘  └─┘┴ ┴└─┘└─┘└─┘
+
+  // dialects that end with a nubmer in their name
+  const specialCases: [Dialect] = ['db2'];
+  for (const specialCase of specialCases) {
+    if (dialect.startsWith(specialCase)) {
+      didEncounterSpecialCase = true;
+      const re: RegExp = new RegExp(`(${specialCase})(.*)`);
+
+      // can use `.match()!` (With bang) b/c we know `re` will match based on `if` condition
+      [_fullString, dialectName, dialectVersion] = dialect.match(re)!;
+      break;
+    }
+  }
+
+  //  \│/   ┌┐┌┌─┐┌┬┐┬┬  ┬┌─┐
+  //  ─ ────│││├─┤ │ │└┐┌┘├┤
+  //  /│\   ┘└┘┴ ┴ ┴ ┴ └┘ └─┘
+
+  if (!didEncounterSpecialCase) {
+    // can use `.match()!` b/c we know `dialect` is populated with something and below will capture
+    [_fullString, dialectName, dialectVersion] = dialect.match(/([a-z-]+)(.*)/)!;
+
+    // Example: "postgres-native" -> "postgres"; "mysql-native" -> "mysql" (REM: not yet implemented)
+    dialectName = dialectName.replace(/(.+?)-native/, '$1');
+  }
+
+  // TypeScript isn't intelligent enough to realize this will be assigned
+  dialectName = dialectName!;
+  dialectVersion = dialectVersion!;
+
+  const defaults: { [key: string]: any } = configs;
+  const config = defaults[dialectName].default as Options;
+  const versions = Object.keys(defaults[dialectName]).filter(v => !/(?:default|native)/.test(v));
```

Comment:
>This whole special case thing can be avoided 

Agreed.  There are other schemes, it's just much better DX to type `DIALECT=mysql8` over `DIALECT=mysql-8` or `DIALECT=mysql DIALECT_VERSION=8` -- lots of us don't use _zsh_.

> if you name the env values postgres-oldest and postgres-latest instead of postgres10 and postgres12 

Yes, it can be avoided, but the goal is to be able to be able to accommodate ad-hoc development containers and be able to develop/test against targeted versions, not just the two that are present.

How do you ensure patches are tested?  As specific minor/patch versions have flaws, how do you ensure they're tested with changes? What's the regression testing process like if you support v10 and v18 and there's a v14-16 issue? -- I'd think you'd include them in the development environment to ensure all developers are capable of testing or running tests against them so that patches are maintained, and bugs aren't re-introduced.  Without versions between oldest-latest, you're not going to catch those flaws.

I'm wondering if you have a more preferred solution other than 2-defaults.  I agree maintaining the _package.json_ would be cumbersome, especially for every version.  What do you think of this? 

Forego environment variable entirely in favor of command-line arguments?
```
Format:
yarn test <dialect> <version>

Example:
yarn test mssql latest
yarn test mssql 2022
yarn test postgres native
```

Instead of any version or dialect being stored in the package.json, create a controller and a script in dev:

package.json
```
  "scripts": {
    "dialects": "yarn ts-node dev/dialects.ts",
    "test": "run(){ SEQ_DIALECT=\"${1}\" SEQ_DIALECT_VERSION=\"${2}\"; yarn prepare &&  ... }; run",
    "test-integration": "run(){ SEQ_DIALECT=\"${1}\" SEQ_DIALECT_VERSION=\"${2}\"; yarn prepare && ... }; run",
    "start": "db(){ SEQ_DIALECT=\"${1:}\"; SEQ_DIALECT_VERSION=\"${2:-latest}\"; $0 \"dev/start.js\" 2>/dev/null; }; db"
  }
```
- remove all the database-specific commands
- _start.js_ would be a controller/router to spin up the respective _/dev/dialects/<db>/<version>_
- no need to edit package.json
- dialects would return supported dialects
- `start` would also show supported dialects if one wasn't entered
- versions can be an enum: oldest, latest, native, <semver>